src/Carbon/Carbon.php

```diff
@@ -1576,6 +1576,16 @@ public function isLeapYear()
     }
 
     /**
+     * Determines if the instance is a long year
+     *
+     * @return bool
+     */
+    public function isLongYear()
+    {
+        return static::create($this->year, 12, 28, 0, 0, 0, $this->tz)->weekOfYear === 53;
```

Comment:
Occasionally, December 31st falls on week 1 of the following year.

For example, when I make a Carbon for 12-31-2018, it's considered week 1. So we wouldn't be able to tell if there were 53 weeks in 2018 by using the 31st.

```
[9] > $date = Carbon\Carbon::create(2018, 12, 31, 0, 0, 0, 'America/New_York');
// object(Carbon\Carbon)(
//   'date' => '2018-12-31 00:00:00.000000',
//   'timezone_type' => 3,
//   'timezone' => 'America/New_York'
// )
[10] > $date->format('W');
// '01'
```

The 28th always falls on week 52 or if it's a long year, week 53. Reference: https://en.wikipedia.org/wiki/ISO_8601#Week_dates